ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -1281,6 +1282,12 @@ void Matmul::validate(
                 TT_FATAL(
                     program_config.per_core_N % program_config.out_subblock_w == 0,
                     "per_core_N must be divisible by out_subblock_w");
+                TT_FATAL(
+                    (program_config.out_subblock_w * program_config.out_subblock_h) <= MAX_HALF_DST_MODE_DST_REG_COUNT,
+                    "out_subblock_w {} times out_subblock_h {} needs to be at most {} to fit in hardware",
+                    program_config.out_subblock_w,
+                    program_config.out_subblock_h,
+                    MAX_HALF_DST_MODE_DST_REG_COUNT);
```

Comment:
for fp32 accum mode, we only use 4 dest regs for half sync mode, and 8 for full sync mode

Reply:
Added logic for this scenario.